src/odemis/driver/zeiss.py

```diff
@@ -0,0 +1,1010 @@
+            if not future._was_stopped:
+                logging.debug("Cancelling failed")
+            return future._was_stopped
+
+    def _createFuture(self):
+        """
+        Return (CancellableFuture): a future that can be used to manage a move
+        """
+        f = CancellableFuture()
+        f._moving_lock = threading.Lock()  # taken while moving
+        f._must_stop = threading.Event()  # cancel of the current future requested
+        f._was_stopped = False  # if cancel was successful
+        f.task_canceller = self._cancelCurrentMove
+        return f
+
+    @isasync
+    def moveRel(self, shift):
+        """
+        shift (dict): shift in m
+        """
+        if not shift:
+            return model.InstantaneousFuture()
+        self._checkMoveRel(shift)
+        foc = self.parent.GetFocus()  #  mm
+        foc += shift["z"] * 1e3
+
+        f = self._createFuture()
+        f = self._executor.submitf(f, self._doMoveAbs, f, foc)
+        return f
+
+    @isasync
+    def moveAbs(self, pos):
+        """
+        pos (dict): pos in m
+        """
+        if not pos:
+            return model.InstantaneousFuture()
+        self._checkMoveAbs(pos)
+
+        foc = pos["z"] * 1e3
+        f = self._createFuture()
+        f = self._executor.submitf(f, self._doMoveAbs, f, foc)
+        return f
+
+    def stop(self, axes=None):
+        """
+        Stop the last command
+        """
+        # Empty the queue (and already stop the stage if a future is running)
+        self._executor.cancel()
+
+        # Try to stop the stage, even if no future is running, for safety
+        logging.warning("Stopping all axes: %s", ", ".join(self.axes))
+        self.parent.Abort()
+
+        try:
+            self._updatePosition()
+        except Exception:
+            logging.exception("Unexpected failure when updating position")
+
+
+class RemconSimulator(object):
+    """
+    Simulates a Keithley 6485
+    Same interface as the serial port
+    """
+
+    def __init__(self, timeout=1, *args, **kwargs):
+        self.timeout = timeout
+        self._output_buf = ""  # what the commands sends back to the "host computer"
+        self._input_buf = ""  # what we receive from the "host computer"
+
+        self._speed = 10  # um/s
+        self._hfw_nomag = 1  # m
+        self.eol = '\r\n'
+
+        # Initialize parameters
+        self.magnification = 10
+        self.horizontalFoV = 5
+        self.pixelSize = 5
+        self.blanker = 0
+        self.external = 0
+        self.posX = 20
+        self.posY = 20
+        self.posZ = 5
+        self.focus = 0
+        self.dur = 0
+        # Prepare moving thread
+        self.target_posX, self.target_posY, self.target_posZ = 0, 0, 0
+        self._end_move = 0
+        self._stage_stop = threading.Event()
+        self.waiting = False
+        
+    def _run_move(self):
+        dur = self._end_move - time.time()
+        while self._stage_stop.wait(dur):
```

Comment:
This `while` seems to be waiting for the wrong thing. It should keep doing this code as long as the current time is before `._end_move`. Inside the loop, instead of putting a (small) sleep, using `self._stage_stop.wait(0.01)`, and immediatly stop in the case the stage stop is aborted.